proto/storage/signature_integration.proto

```diff
@@ -19,3 +21,28 @@ message CosignPublicKeyVerification {
   }
   repeated PublicKey public_keys = 3;
 }
+
+// Holds all verification data for verifying certificates attached to cosign signatures.
+// In case both the certificate and chain are given, both will be used to verify the public key
+// as well as the chain.
+// If only the certificate is given, the Fulcio trusted root chain will be assumed and verified against.
+// If only the chain is given, this will be used over the Fulcio trusted root chain for verification.
+// If no certificate or chain is given, the Fulcio trusted root chain will be assumed and verified against.
+message CosignCertificateVerification {
+  // PEM encoded certificate chain to use for verification.
+  string certificate_chains_pem_enc = 1;
```

Comment:
nit: kinda feel like the prefixing every field with `certificate_` is redundant based on the message name.

Reply:
While the message context is there, I'd rather keep the prefix when using e.g. JSON via our grpc gateway where the message name might not be as clear.